game-app/game-core/src/main/java/games/strategy/engine/framework/startup/mc/ClientModel.java

```diff
@@ -147,13 +147,20 @@ public void doneSelectingPlayers(final byte[] gameData, final Map<String, INode>
         }
       };
 
+  private final Runnable showMainFrame;
+  private final Runnable clientLeftGame;
+
   public ClientModel(
       final GameSelectorModel gameSelectorModel,
-      final SetupPanelModel typePanelModel,
-      final LaunchAction launchAction) {
+      final ServerSetupModel typePanelModel,
+      final LaunchAction launchAction,
+      final Runnable showMainFrame,
+      final Runnable clientLeftGame) {
     this.launchAction = launchAction;
     this.typePanelModel = typePanelModel;
     this.gameSelectorModel = gameSelectorModel;
+    this.showMainFrame = showMainFrame;
+    this.clientLeftGame = clientLeftGame;
```

Comment:
`ClientModel` is a class where I'm not sure if it should be part of the headed subproject.
It clearly is only used by the headed client, but it has an odd relationship with `ServerModel` which seems to suggest there is logic in here that's independent of UI, which might indicate the need for s separate class that should be kept in game-core.
Due to this nature this class is also connected with other parts of the code that would need to be carefully separated before being able to move this class to the headed subproject.